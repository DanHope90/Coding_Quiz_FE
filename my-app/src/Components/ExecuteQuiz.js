import React, { useState } from "react";
import PropTypes from "prop-types";
import QuestionCard from "./QuestionCard";
import "../Styles/ExecuteQuiz.css";

function ExecuteQuiz(props) {
  const { desiredQuiz } = props;

  const [score, setScore] = useState(0);
  const [qNumber, setQNumber] = useState(0);

  const [chosenAnswer, setChosenAnswer] = useState("");

  return (
    <div>
      {(((qNumber + 1) <= desiredQuiz.questions.length) && (
        <>
          <h1>{desiredQuiz.quizName}</h1>
          <h2>{desiredQuiz.description}</h2>
          <div className="question-count">
            Current Question: {qNumber + 1} of {desiredQuiz.questions.length}.
          </div>
          <div className="question-count">
            Your score: {score} out of {desiredQuiz.questions.length}.
          </div>
          <div className="question-card">
            <QuestionCard
              score={score}
              setScore={setScore}
              qNumber={qNumber}
              setQNumber={setQNumber}
              chosenAnswer={chosenAnswer}
              setChosenAnswer={setChosenAnswer}
              question={desiredQuiz.questions[qNumber]}
            />
          </div>

        </>
      )) || (
        <div>
          You have scored {score} out of {desiredQuiz.questions.length}!
        </div>
      )}
    </div>
  );
}

export default ExecuteQuiz;

ExecuteQuiz.propTypes = {
  desiredQuiz: PropTypes.shape({
    quizName: PropTypes.string,
    description: PropTypes.string,
    questions: PropTypes.arrayOf({
      question: PropTypes.string,
      correct_answer: PropTypes.string,
      answers: PropTypes.arrayOf({
        answer: PropTypes.string,
      }),
    }),
  }),
}.isRequired;
