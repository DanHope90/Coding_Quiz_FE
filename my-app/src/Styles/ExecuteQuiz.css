button {
    flex-wrap: wrap;
    border-radius: 12px;
    color: white;
    background-color: black;
    width: 400px;
    height: 75px;
}

button:hover{
    color:black;
    background-color: white;
}

button:active {
    color: red;
    background-color: white;
}

.quiz-body {
    display: flex;
    flex-direction: column;
    gap: 5px;
    align-items: center;
}

h1 {
    font-size: 24px;
}

h2 {
    font-size: 18px;
}

.question-count {
    font-size: 16px;
    margin: 10px;
    color: blue;
    font-weight: bold;

}

.question-card {
    font-size: 16px;
}

.final-score {
    padding-bottom: 20px;
    font-weight: bold;
}

.leave-random-quiz {
    display: grid;
    justify-items: center;
    color: red;
}

@media only screen and (max-width: 650px) {
    button {
        width: 250px;
        height: 37.5px;
        font-size: 8px;
    }
    h1 {
        font-size: 20px;
    }
    h2 {
        font-size: 15px;
    }
    .question-count {
        font-size: 12px;
    }
    .question-card {
        font-size: 8px;
    }
    .final-score {
        font-size: 12px;
    }
}