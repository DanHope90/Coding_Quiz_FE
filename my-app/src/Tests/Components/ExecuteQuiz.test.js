import React from "react";
import renderer from "react-test-renderer";
import ExecuteQuiz from "../../Components/ExecuteQuiz";

test("renders the ExecuteQuiz component correctly", () => {
  const testRenderer = renderer.create(<ExecuteQuiz />);
  expect(testRenderer).toMatchSnapshot();
});
