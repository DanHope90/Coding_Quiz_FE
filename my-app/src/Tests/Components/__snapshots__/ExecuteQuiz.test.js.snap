// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`renders the ExecuteQuiz component correctly 1`] = `
<div>
  <div>
    You have scored 
    0
     out of 
    0
    !
  </div>
</div>
`;
